### ISSUE TITLE
Building project with Yarn 3 Plug'n'Play (PnP) fails

### ISSUE THREAD

**Issue by jensborje** [strapi/strapi#16373]
## Bug report
Building project fails when using Yarn 3 with Plug'n'Play (pnp) enabled.

### Required System information

- Node.js version: `v16.18.1`
- Yarn version: `3.5.0`
- Strapi version: `4.9.1`
- Database: `sqlite`
- Operating system: `Mac OS`
- Is your project Javascript or Typescript: `Javascript`

### Describe the bug

Building project using Yarn 3 with Plug'n'Play (pnp) enabled fails due to dependency `react-router` is not declared in `@strapi/admin`.

```
This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). The promise rejected with the reason:
Error: @strapi/admin tried to access react-router, but it isn't declared in its dependencies; this makes the require call ambiguous and unsound.
```

Also tried `pnpMode: loose` to no use but different errors.

### Steps to reproduce the behavior

1. Install fresh project
2. Set yarn version to 3 `yarn set version berry`
3. Set `nodeLinker: pnp` in `yarnrc.yml`
4. Run `yarn install`
5. Run `yarn build`
6. See error

### Expected behavior

The build should succeed allowing us to use Yarn pnp with zero-installs.

**Comment by joshuaellis** [comment]
Sorry to see this, this is a silly mistake to make. I've opened up a PR to resolve this issue, when it's merged you should be able to use the `next` version on NPM the day after (we release nightly vesions)

**Comment by jensborje** [comment]
Thank you for your fast response.
Unfortunately, I'm still experience this issue running the latest version with `next` tag.

The stack trace points to 
```
@strapi-admin-npm-0.0.0-next.9542f2748460d5d94905e89d1902c0e1625fbea0-8a3eb1aa69-7fca89fca4.zip/node_modules/@strapi/admin/webpack.alias.js
```

I can see that it is indeed listed there, [as pointed out in the PR](https://github.com/strapi/strapi/pull/16375#issuecomment-1506709734).

If I'm missing any required step for this to work, I'm eager to hear about it!

**Comment by csharpee** [comment]
I have the same error when installing with yarn 3.3.1

When I run **yarn develop**, the following error occurs.
```
This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). The promise rejected with the reason:
Error: @strapi/admin tried to access react-router, but it isn't declared in its dependencies; this makes the require call ambiguous and unsound.

Required package: react-router
Required by: @strapi/admin@virtual:fecea0005c12c3c685ba9c4428d2780c57a11815062455a43b19689f77fea3226f3495a24aff0ddc433706f5b5d55741f8f6727ae5f86012113b9b6f246a4124#npm:4.9.2 (via /home/csharpe/Desktop/Local-Development/Strapi/strapi-dev/.yarn/__virtual__/@strapi-admin-virtual-966872b6d9/0/cache/@strapi-admin-npm-4.9.2-b6333a3952-290e3d861b.zip/node_modules/@strapi/admin/)

```

I also had to `yarn install` after doing the **--quickstart** option, but after running yarn install, it installed successfully.

**Comment by ddresch** [comment]
Need to put myself in the line of me toos. Sorry, but no offense, messing around for more than 7 hours to get bare Strapi up and running is not a good reputation for the project. Thanks for the hint of @csharpee who pointed to the contributing document. Doing a fork and clone the repo, you can get a running version. 

I tested the different node versions, starting with 14 up to 18. First I messed around because I thought it's an issue of using pnpm which is definitely not working with Strapi. So use yarn – got it! :)

I'm on Mac M1, so I guess there is an issue with some prebuild dependencies, not sure. Using `nvm` as node manager also seems an issue. Did an installation on a fresh Ubuntu 22.04 box and a node 16.20 installation (no nvm). All working! 

I'm not sure what I can contribute to the issue, but it's undoubtedly something worth investigating further. My env setup is nothing special using nvm is also pretty common.

**Comment by yasssuz** [comment]
Same issue here on `@strapi/strapi: 4.10.2` (lastest version).
I get this error: 
```
This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). The promise rejected with the reason:
strapi:dev: Error: @strapi/admin tried to access react-router, but it isn't declared in its dependencies; this makes the require call ambiguous and unsound.
```
---
Currently, the workaround is to create a `.yarnrc.yml` file in the root of the project and add this line: 
```
nodeLinker: node-modules
```

> By doing this you will disable PnP completely and fallback to a `node_modules` folder, but at least the production build can keep running.

**Comment by joshuaellis** [comment]
Should be fixed in `4.10.3`
